packages/server/src/api/routes/tests/viewV2.spec.ts

```diff
@@ -867,157 +846,334 @@ describe.each([
-    })
 
-    it("can delete an existing view", async () => {
-      const tableId = table._id!
-      const getPersistedView = async () =>
-        (await config.api.table.get(tableId)).views![view.name]
+      it("does not include columns removed from the table", async () => {
+        const view = await config.api.viewV2.create({
+          tableId,
+          name: generator.guid(),
+          schema: {
+            id: { visible: true },
+            one: { visible: true },
+            two: { visible: true },
+          },
+        })
+        const table = await config.api.table.get(tableId)
+        const { one: _, ...newSchema } = table.schema
+        await config.api.table.save({ ...table, schema: newSchema })
+
+        expect(await getDelegate(view)).toEqual({
+          ...view,
+          schema: {
+            id: { ...table.schema["id"], visible: true },
+            two: { ...table.schema["two"], visible: true },
+            three: { ...table.schema["three"], visible: false },
+          },
+        })
+      })
 
-      expect(await getPersistedView()).toBeDefined()
+      it("does not include columns hidden from the table", async () => {
+        const view = await config.api.viewV2.create({
+          tableId,
+          name: generator.guid(),
+          schema: {
+            id: { visible: true },
+            one: { visible: true },
+            two: { visible: true },
+          },
+        })
+        const table = await config.api.table.get(tableId)
+        await config.api.table.save({
+          ...table,
+          schema: {
+            ...table.schema,
+            two: { ...table.schema["two"], visible: false },
+          },
+        })
 
-      await config.api.viewV2.delete(view.id)
+        expect(await getDelegate(view)).toEqual({
+          ...view,
+          schema: {
+            id: { ...table.schema["id"], visible: true },
+            one: { ...table.schema["one"], visible: true },
+            three: { ...table.schema["three"], visible: false },
+          },
+        })
+      })
 
-      expect(await getPersistedView()).toBeUndefined()
-    })
-  })
+      it("should be able to fetch readonly config after downgrades", async () => {
+        mocks.licenses.useViewReadonlyColumns()
+        const res = await config.api.viewV2.create({
+          name: generator.name(),
+          tableId: table._id!,
+          schema: {
+            id: { visible: true },
+            one: { visible: true, readonly: true },
+          },
+        })
 
-  describe("fetch view (through table)", () => {
-    it("should be able to fetch a view V2", async () => {
-      const res = await config.api.viewV2.create({
-        name: generator.name(),
-        tableId: table._id!,
-        schema: {
-          id: { visible: true },
-          Price: { visible: false },
-          Category: { visible: true },
-        },
+        mocks.licenses.useCloudFree()
+        const view = (await getDelegate(res)) as ViewV2
```

Comment:
`table.views` still contains `View | ViewV2`. `View` does not have a schema, but for the nature of these tests, it will always be ViewV2. I just changed the cast to be part of the fetch, so it can be removed from the test itself